RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -186,6 +196,15 @@ final class PurchaseHandler: ObservableObject {
         cancellables.removeAll()
     }
 
+    /// Resets the session purchase result for a new paywall session.
+    ///
+    /// This is called when a paywall appears to ensure we track purchases for the current session only.
+    /// Since `PurchaseHandler` is a `@StateObject` that persists, the `sessionPurchaseResult` would
```

Comment:
Not sure this is correct. `PurchaseHandler` is created whenever `PaywallViewConfiguration` is created. So its lifespan is actually linked to the `PaywallView`'s lifespan.
Perhaps I'm not understanding the concept of "session" here 😅

Reply:
Before this we used to have a `purchased` flag which served a similar purpose. 
Session means the lifespan of the PaywallView(Initial) + PaywallView(ExitOffer). Would calling this "PaywallView(Initial) + PaywallView(ExitOffer)" make more sense? I don't want to introduce new naming  😓